src/tools/getDatasourceMetadata/datasourceMetadataUtils.ts

```diff
@@ -29,53 +71,55 @@ export function buildDataDictionary(
     }
 
-    dataDictionary.fields.push(toPush);
+    combinedFields.fields.push(toPush);
   }
 
-  if (
-    !listFieldsResult.data.publishedDatasources ||
-    listFieldsResult.data.publishedDatasources.length === 0 ||
-    !listFieldsResult.data.publishedDatasources[0].fields
-  ) {
-    return dataDictionary;
+  if (!listFieldsResult.data.publishedDatasources[0]?.fields.length) {
+    return combinedFields;
   }
 
-  // Of the fields in our dataDictionary, populate them with additional properties we get from listFields results.
-  for (const field of dataDictionary.fields) {
-    const matchingField = listFieldsResult.data.publishedDatasources[0].fields.find(
-      (f: any) => f.name === field.name,
+  // Of the fields in our response object, populate them with additional properties we get from listFields results.
+  for (const field of combinedFields.fields) {
+    const matchingListField = listFieldsResult.data.publishedDatasources[0].fields.find(
+      (f) => f.name === field.name,
     );
-    if (!matchingField) {
-      continue;
+    if (matchingListField) {
+      populateFieldWithAdditionalProperties(matchingListField as Field, field);
     }
-    if (matchingField.description) {
-      field.description = matchingField.description;
-    }
-    if (matchingField.descriptionInherited && matchingField.descriptionInherited.length > 0) {
-      field.descriptionInherited = matchingField.descriptionInherited;
-    }
-    if (matchingField.dataCategory) {
-      field.dataCategory = matchingField.dataCategory;
-    }
-    if (matchingField.role) {
-      field.role = matchingField.role;
-    }
-    if (matchingField.defaultFormat) {
-      field.defaultFormat = matchingField.defaultFormat;
-    }
-    if (matchingField.formula && matchingField.formula.length > 0) {
-      field.formula = matchingField.formula;
-      if (matchingField.isAutoGenerated != undefined) {
-        field.isAutoGenerated = matchingField.isAutoGenerated;
-      }
-      if (matchingField.hasUserReference != undefined) {
-        field.hasUserReference = matchingField.hasUserReference;
-      }
+  }
+
+  return combinedFields;
+}
+
+function populateFieldWithAdditionalProperties(sourceField: Field, targetField: Field): void {
+  if (sourceField.description?.length) {
+    targetField.description = sourceField.description;
+  }
+  if (sourceField.descriptionInherited?.length) {
+    targetField.descriptionInherited = sourceField.descriptionInherited;
+  }
+  if (sourceField.dataCategory) {
+    targetField.dataCategory = sourceField.dataCategory;
+  }
+  if (sourceField.role) {
+    targetField.role = sourceField.role;
+  }
+  if (sourceField.defaultFormat) {
+    targetField.defaultFormat = sourceField.defaultFormat;
+  }
+  if (sourceField.formula?.length) {
+    targetField.formula = sourceField.formula;
+    // Possibly null or undefined, only populate if true or false.
+    if (sourceField.isAutoGenerated != undefined) {
```

Comment:
If you really wanted to bypass the wonky JavaScript null/undefined equality business:

```ts
if ([true, false].includes(sourceField.isAutoGenerated))
```